### PR TITLE
Add ColorEditor tests and fix some bugs

### LINKED ISSUE #1290: ColorEditor.EditValue throws InvalidCastException if `value` is not a Color
Should handle this gracefully as if it were null

```cs
public static IEnumerable<object[]> EditValue_TestData()
{
    yield return new object[] { null };
    yield return new object[] { "value" };
    yield return new object[] { Color.Empty };
    yield return new object[] { Color.Red };
    yield return new object[] { new object() };
}

[Theory]
[MemberData(nameof(EditValue_TestData))]
public void ColorEditor_EditValue_ValidProvider_ReturnsValue(object value)
{
    var editor = new ColorEditor();
    var mockEditorService = new Mock<IWindowsFormsEditorService>(MockBehavior.Strict);
    var mockServiceProvider = new Mock<IServiceProvider>(MockBehavior.Strict);
    mockServiceProvider
        .Setup(p => p.GetService(typeof(IWindowsFormsEditorService)))
        .Returns(mockEditorService.Object)
        .Verifiable();
    mockEditorService
        .Setup(e => e.DropDownControl(It.IsAny<Control>()))
        .Verifiable();
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Once());
    mockEditorService.Verify(e => e.DropDownControl(It.IsAny<Control>()), Times.Once());

    // Edit again.
    Assert.Equal(value, editor.EditValue(null, mockServiceProvider.Object, value));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
    mockServiceProvider.Verify(p => p.GetService(typeof(IWindowsFormsEditorService)), Times.Exactly(2));
}
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #1290
Contributes to #817

### CODE REVIEW

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs [hunk 1]
```diff
@@ -17,40 +17,37 @@
...
 
         /// <summary>
         /// Edits the given object value using the editor style provided by ColorEditor.GetEditStyle.
         /// </summary>
         [SuppressMessage("Microsoft.Performance", "CA1808:AvoidCallsThatBoxValueTypes")]
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
-            object returnValue = value;
-
             if (provider != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs]: Please invert if to reduce nesting

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs [hunk 2]
```diff
@@ -17,40 +17,37 @@
...
             {
-                IWindowsFormsEditorService edSvc = (IWindowsFormsEditorService)provider.GetService(typeof(IWindowsFormsEditorService));
-
-                if (edSvc != null)
```
Comment (RussKie) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs]: ditto, and the next line can be reverted

**Comment by hughbe** [comment]
> Can you please clarify what the bug was?

If you called `new ColorEditor().EditValue(null, provider, new object())` then we'd throw an `InvalidCastException`

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs [hunk 3]
```diff
@@ -17,43 +17,42 @@
...
         public override object EditValue(ITypeDescriptorContext context, IServiceProvider provider, object value)
         {
-            object returnValue = value;
-
-            if (provider != null)
+            if (provider == null)
+            {
+                return value;
+            }
+            if (!(provider.GetService(typeof(IWindowsFormsEditorService)) is IWindowsFormsEditorService edSvc))
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs]: Contribution to #817

File: src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs [hunk 4]
```diff
@@ -17,43 +17,42 @@
...
-                    if (colorUI.Value != null && ((Color)colorUI.Value) != Color.Empty)
-                    {
-                        value = colorUI.Value;
-                    }
+            _colorUI.Start(edSvc, value);
+            edSvc.DropDownControl(_colorUI);
 
-                    colorUI.End();
-                }
+            if (_colorUI.Value is Color colorValue && colorValue != Color.Empty)
```
Comment (hughbe) [inline src/System.Windows.Forms.Design.Editors/src/System/Drawing/Design/ColorEditor.cs]: Fix for #1290

**Pull request merged**